doc/en/explanation/goodpractices.rst

```diff
@@ -12,41 +12,26 @@ For development, we recommend you use :mod:`venv` for virtual environments and
 as well as the ``pytest`` package itself.
 This ensures your code and dependencies are isolated from your system Python installation.
 
-Next, place a ``pyproject.toml`` file in the root of your package:
+Create a ``pyproject.toml`` file in the root of your repository as described in
+:doc:`packaging:tutorials/packaging-projects`.
+The first few lines should look like this:
 
 .. code-block:: toml
 
     [build-system]
-    requires = ["setuptools>=42", "wheel"]
-    build-backend = "setuptools.build_meta"
-
-and a ``setup.cfg`` file containing your package's metadata with the following minimum content:
-
-.. code-block:: ini
+    requires = ["hatchling"]
```

Comment:
hmm, maybe we can also use tabs like in [the packaging tutorial][]?

I don’t recommend setuptools as it’s:

- slow
- messy (leaves a .egg-info folder lying around in your workspace, emits a bunch of unnecessary log messages during build)
- hard to configure (a bunch of legacy config files like MANIFEST.in, and misleadingly named config keys)
- no longer the first choice on [the packaging tutorial][]

but of course I can leave it if you prefer. I’d still get rid of `setup.cfg` since setuptools these days [supports PEP 621](https://setuptools.pypa.io/en/latest/userguide/quickstart.html). Unfortunately configuring it with more than just basic metadata is labelled as “beta”, e.g. this:

```toml
[tool.setuptools.packages.find]
where = ["src"]  # ["."] by default
```

[the packaging tutorial]: https://packaging.python.org/en/latest/tutorials/packaging-projects/

Reply:
> no longer the first choice on [the packaging tutorial](https://packaging.python.org/en/latest/tutorials/packaging-projects/)

Ahh thanks for pointing that out. If the packaging tutorial recommends `hatchling` as first choice, I think we should follow suit then.

Reply:
In case you would like some context, I followed that discussion, and the reasoning went something like that:

1. we don’t want to promote any standards following build backend
2. so we use tabs
3. but since the first (auto-selected) tab is kind of an endorsement, we need something there that is good for the vast majority of people, i.e. pure python packages
4. because of the above mentioned drawbacks of setuptools, the first tab will be hatchling, because it provides a better experience for people creating a pure python package than setuptools

Hatchling doesn’t have a sufficient amount of plugins or example code yet to allow a highly custom build like numpy or numba or so, but for anything with purely static build configuration, it’s a very good choice.